test/cpp/common/alts_context_util_test.cc

```diff
@@ -0,0 +1,189 @@
+      experimental::GetAltsContextFromAuthContext(auth_context);
+  EXPECT_NE(alts_context.get(), nullptr);
+  EXPECT_EQ(expected_ap, alts_context->application_protocol());
+  EXPECT_EQ(expected_rp, alts_context->record_protocol());
+  EXPECT_EQ(expected_peer, alts_context->peer_service_account());
+  EXPECT_EQ(expected_local, alts_context->local_service_account());
+  EXPECT_EQ(expected_sl, alts_context->security_level());
+  // all rpc versions should be 0 if not set
+  experimental::AltsContext::RpcProtocolVersions rpc_protocol_versions =
+      alts_context->peer_rpc_versions();
+  EXPECT_EQ(0, rpc_protocol_versions.max_rpc_version.major_version);
+  EXPECT_EQ(0, rpc_protocol_versions.max_rpc_version.minor_version);
+  EXPECT_EQ(0, rpc_protocol_versions.min_rpc_version.major_version);
+  EXPECT_EQ(0, rpc_protocol_versions.min_rpc_version.minor_version);
+}
+
+TEST(AltsContextUtilTest, AuthContextWithGoodAltsContext) {
+  grpc_core::RefCountedPtr<grpc_auth_context> ctx =
+      grpc_core::MakeRefCounted<grpc_auth_context>(nullptr);
+  SecureAuthContext auth_context(ctx.get());
+  ctx.reset();
+
+  upb::Arena context_arena;
+  grpc_gcp_AltsContext* context = grpc_gcp_AltsContext_new(context_arena.ptr());
+  upb::Arena versions_arena;
+  grpc_gcp_RpcProtocolVersions* versions =
+      grpc_gcp_RpcProtocolVersions_new(versions_arena.ptr());
+  upb::Arena max_major_version_arena;
+  grpc_gcp_RpcProtocolVersions_Version* version =
+      grpc_gcp_RpcProtocolVersions_Version_new(max_major_version_arena.ptr());
+  grpc_gcp_RpcProtocolVersions_Version_set_major(version, 10);
+  grpc_gcp_RpcProtocolVersions_set_max_rpc_version(versions, version);
+  grpc_gcp_AltsContext_set_peer_rpc_versions(context, versions);
+  size_t serialized_ctx_length;
+  char* serialized_ctx = grpc_gcp_AltsContext_serialize(
+      context, context_arena.ptr(), &serialized_ctx_length);
+  EXPECT_NE(serialized_ctx, nullptr);
+  auth_context.AddProperty(TSI_ALTS_CONTEXT,
+                           string(serialized_ctx, serialized_ctx_length));
+  std::unique_ptr<experimental::AltsContext> alts_context =
+      experimental::GetAltsContextFromAuthContext(auth_context);
+  EXPECT_NE(alts_context.get(), nullptr);
+  EXPECT_EQ("", alts_context->application_protocol());
+  EXPECT_EQ("", alts_context->record_protocol());
+  EXPECT_EQ("", alts_context->peer_service_account());
+  EXPECT_EQ("", alts_context->local_service_account());
+  EXPECT_EQ(GRPC_SECURITY_NONE, alts_context->security_level());
+  experimental::AltsContext::RpcProtocolVersions rpc_protocol_versions =
+      alts_context->peer_rpc_versions();
+  EXPECT_EQ(10, rpc_protocol_versions.max_rpc_version.major_version);
+  EXPECT_EQ(0, rpc_protocol_versions.max_rpc_version.minor_version);
+  EXPECT_EQ(0, rpc_protocol_versions.min_rpc_version.major_version);
+  EXPECT_EQ(0, rpc_protocol_versions.min_rpc_version.minor_version);
+}
+
+TEST(AltsContextUtilTest, AltsClientAuthzCheckFunction) {
```

Comment:
nit: Add a test that AltsContext does not exist in AuthContext, expect permission denied.